docs/developer-docs/functionality/sns/predeployment.md

```diff
@@ -1 +1,182 @@
+   initial neurons voting in favor of this proposal.
+   The initial neurons cannot do other things, such as changing the SNS
+   parameters, as the SNS governance canister is still in pre-genesis mode.
+
+4) **Ask the IC to start the decentralization sale**: The decentralization sale
+   is started by an NNS proposal that can be submitted by anyone and is decided on by the
+   IC community. This means that effectively you hand over the control of your dapp
+   to the SNS and ask the IC to decentralize the SNS in a decentralization sale.
+   When voting on a NNS proposal to start a decentralization sale, the NNS neurons can check
+   the sale parameters and whether the dapp’s control has been handed over to the SNS.
+   The NNS proposal thus also serves as a safeguard where the wisdom of the crowd can
+   detect potentially malicious SNSs before they trick users into investing in them.
+   If the NNS proposal is adopted, the sale is started. If the NNS proposal is rejected,
+   the SNS launch is aborted and the dapp’s control is handed back to you, i.e., to the 
+   original developers of the dapp.
+
+5) **Decentralization sale**: When the sale starts, the sale canister holds the number
+   of SNS tokens that were specified at initialization (see Step 1). End users can
+   participate in the decentralization sale by transferring ICP tokens to the 
+   sale canister.
+
+6) **SNS genesis**: When the decentralization sale ends, it is first established whether
+   it was successful, e.g., enough ICP have been collected. If the sale was successful,
+   the exchange rate is determined and all tokens are given to the sale participants in
+   neurons. Once all neurons are created, the SNS is under decentralized control
+   and the governance canister is set to be fully functional.
+   If the sale is not successful, the decentralization attempt failed and everything
+   is reverted to the state before the SNS launch, including that the dapp’s control
+   is handed back to you (i.e., the original developers of the dapp), and the 
+   collected ICP are refunded to the sale participants.
+
+## Setting the SNS parameters
+As mentioned above, the first step in launching an SNS is to choose the parameters that
+your SNS will be launched with.
+This lets you choose many SNS settings and also lets you choose the initial 
+token distribution.
+
+You can set all these parameters in a _yaml_ file that can then be passed as an argument
+when creating the SNS, both for testing and in production.
+There are some parameters that you have to actively set and others that are set to a
+default value but that you can also change if you like.
+To make sure that all parameters are set to valid values, that are also consistent
+with each other, you can use a tool that validates your input file. 
+
+**To create the SNS parameter yaml file and validate it, follow the steps
+[Defining the initial parameters for the SNS]().**
+// todo : add link
```

Comment:
is this TODO tracked somehow?

Reply:
It is not, but I have changed this now to be in a comment as in the other places.
I then have one todo on my list to add the links to all the documentation in the end (the required tools and thus their documentation are still being worked on, so I think it makes sense to add this later in one go)